interface/src/app/node/high_weight_genes.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Test <high-weight-genes> directive.
+ **/
+
+describe('<high-weight-genes> directive', function() {
+  beforeEach(module('adage.node'));
+
+  // Thanks to "grunt-html2js" module (called by Grunt), the template file
+  // "node/high_weight_genes.tpl.html" has been converted into an Angular
+  // module of the same name and put in $templateCache before karma runs.
+  beforeEach(module('node/high_weight_genes.tpl.html'));
+
+  var $compile, $httpBackend, $rootScope;
+  beforeEach(inject(function(_$compile_, _$httpBackend_, _$rootScope_) {
+    $compile = _$compile_;
+    $httpBackend = _$httpBackend_;
+    $rootScope = _$rootScope_;
+  }));
+
+  it('should render HTML correctly', function() {
+    var parentScope = $rootScope.$new();
+    parentScope.nodeID = 123;
+    var testHTML = '<high-weight-genes node-id="{{nodeID}}">' +
+        '</high-weight-genes>';
+    var uri = '/api/v0/participation?limit=0&node=' + parentScope.nodeID;
+    var element = $compile(testHTML)(parentScope);
+
+    // Mocked response data of genes:
+    var mockGenes = [
+      {gene: {
+        'systematic_name': 'g1_sys',
+        'standard_name': 'g1_std',
+        'description': 'g1_desc'
+      }},
+      {gene: {
+        'systematic_name': 'g2_sys',
+        'standard_name': 'g2_std',
+        'description': 'hypothetical protein'
+      }},
+      {gene: {
+        'systematic_name': 'g3_sys',
+        'standard_name': 'g3_std',
+        'description': 'g3_desc'
+      }}
+    ];
+
+    $httpBackend.expectGET(uri).respond({objects: mockGenes});
```

Comment:
By the way, some Angular books or tutorials actually call Karma a framework of **unit and integration test**, because of the its ability to test both individual components (controller, service, etc) and the integrated parts (mocked server response, compiled HTML template in directive, etc).